### ISSUE TITLE
Why does the memory increase after smoothquantization?

### ISSUE THREAD

**Issue by runbaozha** [NVIDIA/TensorRT-LLM#139]
I use the smoothquant to quant llama-1B.
Before smoothquant,I run  xxx.py:

device = "cuda"
tokenizer = AutoTokenizer.from_pretrained("saibo/llama-1B", use_fast=False)
model = AutoModelForCausalLM.from_pretrained("saibo/llama-1B")
inputs = tokenizer('who are you?', return_tensors="pt")

model.to(device)
inputs.to(device)

output_ids = model.generate(inputs.input_ids)
output = tokenizer.decode(output_ids[0])

The memory is about 3GB,But when I use smoothquant to quant llama follow the TensorRT-LLM/examples/llama/README.md ,and the memory is about 8GB.
When I build the engin, there are some warnings in the terminal log:

[10/26/2023-08:29:25] [TRT] [W] Missing scale and zero-point for tensor LLaMAForCausalLM/SHUFFLE_18_output_0, expect fall back to non-int8 implementation for any layer consuming or producing given tensor
[10/26/2023-08:29:25] [TRT] [W] Missing scale and zero-point for tensor LLaMAForCausalLM/lm_head/CONSTANT_0_output_0, expect fall back to non-int8 implementation for any layer consuming or producing given tensor
[10/26/2023-08:29:25] [TRT] [W] Missing scale and zero-point for tensor logits, expect fall back to non-int8 implementation for any layer consuming or producing given tensor
...
[TensorRT-LLM][WARNING] Cannot profile configuration 18 (for m=1, n=12288, k=4096). Skipped
[TensorRT-LLM][WARNING] Cannot profile configuration 19 (for m=1, n=12288, k=4096). Skipped
[TensorRT-LLM][WARNING] Cannot profile configuration 22 (for m=1, n=12288, k=4096). Skipped
[TensorRT-LLM][WARNING] Cannot profile configuration 23 (for m=1, n=12288, k=4096). Skipped
[TensorRT-LLM][WARNING] Cannot profile configuration 0 (for m=2, n=12288, k=4096). Skipped
[TensorRT-LLM][WARNING] Cannot profile configuration 18 (for m=2, n=12288, k=4096). Skipped

Did quant the model fail or other reason make the memory increase?

**Comment by jdemouth-nvidia** [comment]
Hi @runbaozha ,

Are you comparing the memory consumption between HuggingFace (without SmoothQuant) and TensorRT-LLM with SmoothQuant? If that's the case, can you tell me if you see a difference in memory between TensorRT-LLM without SmoothQuant and TensorRT-LLM with SmoothQuant, please?

Thanks,
Julien

**Comment by runbaozha** [comment]
> Hi @runbaozha ,
> 
> Are you comparing the memory consumption between HuggingFace (without SmoothQuant) and TensorRT-LLM with SmoothQuant? If that's the case, can you tell me if you see a difference in memory between TensorRT-LLM without SmoothQuant and TensorRT-LLM with SmoothQuant, please?
> 
> Thanks, Julien

Yes, the memory of HuggingFace(without SmoothQuant) and TensorRT-LLM without SmoothQuant are both 3GB,but TensorRT-LLM with SmoothQuant is 8GB. I am confused and want to know what causes, thank you.

**Comment by jdemouth-nvidia** [comment]
For TensorRT-LLM, is it the engine created by TensorRT that is much bigger or the TensorRT-LLM runtime? To be honest, we will have to investigate why it is bigger. We may not have the time to do it right now but we will work on it in the future. Thanks.

**Comment by litaotju** [comment]
@runbaozha could you please share the version and the commands you run TRT-LLM such that we can repro and debug your case? Thx

**Comment by litaotju** [comment]
@runbaozha we had recently fixed serveral bugs related to smoothquant memory usage. Could you try if the latest main branch works for this case. 

I am closing this. If you still see the issues on latest main, please help to update and reopen the issue, and provide commands you are using. Thanks.
